packages/experimental/NativeFontMetrics/src/fontMetrics.ts

```diff
@@ -0,0 +1,23 @@
+import { NativeEventEmitter, Platform } from 'react-native';
+import NativeFontMetrics from './NativeFontMetrics';
+import { IFontMetrics, ScaleFactors } from './NativeFontMetrics.types';
+
+const eventEmitter = new NativeEventEmitter(NativeFontMetrics as any);
+
+class FontMetricsImpl implements IFontMetrics {
+  _scaleFactors: ScaleFactors;
+
+  constructor() {
+    this._scaleFactors = NativeFontMetrics.currentScaleFactors();
+    eventEmitter.addListener('onFontMetricsChanged', ({ newScaleFactors }) => {
+      this._scaleFactors = newScaleFactors;
+    });
+  }
+
+  get scaleFactors(): ScaleFactors {
+    return this._scaleFactors;
+  }
+}
+
+const fontMetricsImpl = Platform.OS === 'ios' ? new FontMetricsImpl() : { scaleFactors: {} };
```

Comment:
Any chance we could separate this into a separate `.ios` file? I think that will result in a smaller bundle size and make it more clear it's iOS only.

Reply:
Done.